tmail-backend/rate-limiter/rate-limiter-cassandra/src/main/scala/com/linagora/tmail/rate/limiter/cassandra/CassandraRateLimitationPlanRepository.scala

```diff
@@ -0,0 +1,70 @@
+package com.linagora.tmail.rate.limiter.cassandra
+
+import com.google.common.base.Preconditions
+import com.linagora.tmail.rate.limiter.api.{LimitTypes, OperationLimitations, RateLimitation, RateLimitationPlanRepository, RateLimitingPlan, RateLimitingPlanCreateRequest, RateLimitingPlanId, RateLimitingPlanName, RateLimitingPlanNotFoundException, RateLimitingPlanResetRequest}
+import com.linagora.tmail.rate.limiter.cassandra.CassandraRateLimitPlanDAO.{LimitingEntry, RateLimitingPlanEntry}
+import org.apache.james.util.ReactorUtils.DEFAULT_CONCURRENCY
+import org.reactivestreams.Publisher
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+import java.time.Duration
+import javax.inject.Inject
+
+class CassandraRateLimitationPlanRepository @Inject()(cassandraRateLimitPlanDAO: CassandraRateLimitPlanDAO) extends RateLimitationPlanRepository {
+
+  override def create(creationRequest: RateLimitingPlanCreateRequest): Publisher[RateLimitingPlan] =
+    SMono.fromCallable(() => RateLimitingPlanId.generate)
+      .flatMap(planId => create(planId, RateLimitingPlanEntry.from(planId, creationRequest)))
+      .flatMap(planId => SMono.fromPublisher(get(planId)))
+
+  private def create(planId: RateLimitingPlanId, insertEntries: Seq[RateLimitingPlanEntry]): SMono[RateLimitingPlanId] =
+    SFlux.fromIterable(insertEntries)
+      .flatMap(insertEntry => cassandraRateLimitPlanDAO.insert(insertEntry))
+      .`then`()
+      .`then`(SMono.just(planId))
+
+  override def update(resetRequest: RateLimitingPlanResetRequest): Publisher[RateLimitingPlan] =
+    SMono.fromPublisher(cassandraRateLimitPlanDAO.planExists(resetRequest.id))
+      .filter(exists => exists)
+      .switchIfEmpty(SMono.error(new RateLimitingPlanNotFoundException))
+      .flatMap(_ => cassandraRateLimitPlanDAO.delete(resetRequest.id))
+      .`then`(create(resetRequest.id, RateLimitingPlanEntry.from(resetRequest)))
+      .`then`(SMono.fromPublisher(get(resetRequest.id)))
```

Comment:
We don't need another read as we do reset things here

Reply:
read for api 
`def update(resetRequest: RateLimitingPlanResetRequest): Publisher[RateLimitingPlan]`

Reply:
Yes but we just read something that we just wrote. We don't need that.